### ISSUE TITLE
replace doLast with dependsOn

### ISSUE THREAD

**Issue by scottdraves** [twosigma/beakerx#5670]
in all our gradle scripts wherever possible.
https://stackoverflow.com/questions/17438219/better-to-use-task-dependencies-or-task-dolast-in-gradle

**Comment by jaroslawmalekcodete** [comment]
I didn't find in our project places where we are using "Alternative A" which I understand can cause problems.  We create separate task for new activities (I'm talking about Alternative B) and I think we are doing it in correct way.
Additionally we don't  use shortcut for 'doLast' method what is recommended by doc https://docs.gradle.org/3.3/userguide/tutorial_using_tasks.html#sec:a_shortcut_task_definition

If I do not understand the problem or I missed something please provide more details

**Comment by altavir** [comment]
I've looked through the gradle files. And it looks OK now. Probably task dependencies were fixed in last update.
The idea is that you should never call one task from another directly (using `doLast` notation), instead use task dependencies. Also it is a good idea to atomize any actions which make any changes to files since some building blocks could be executed in parallel.
The only problematic place I see for now is this one:
```groovy
task staticContent(dependsOn: build) {
  doLast {
    cleanBeakerxStatic(kernelIdName)
    copyLibsToBeakerxStatic(kernelIdName, configurations, jar)
    saveBeakerxStaticKernelJson(kernelIdName, "$mainClassName", mainClassToReplace)
  }
}
```

It is better to create a separate tasks for each of given actions (clean. copy and generate meta) in each kernel and use them. No need to copy functions in each kernel, you can just do something like this:
```geoovy
subprojects{ kernel ->
  task cleanStatic{
    cleanBeakerxStatic(kernel.name)
  }
}
```

**Comment by scottdraves** [comment]
thx for the review altavir.
